spec/integration/data_stream/manifest.spec.yml

```diff
@@ -349,6 +348,15 @@ spec:
                             minProperties: 1
                             additionalProperties:
                               $ref: "#/definitions/elasticsearch_index_template/properties/mappings/properties/dynamic_templates/items/patternProperties/^_embedded_ecs/properties/mapping"
+            subobjects:
+              description: >
+                Specifies if field names containing dots should be expanded into subobjects.
+                For example, if this is set to `true`, a field named `foo.bar` will be
+                expanded into an object with a field named `bar` inside an object
+                named `foo`.
```

Comment:
This makes it sound like it has an effect on the _source of the document, which is not the case.

I think we should also list the limitations when this is set to false. The primary limitation is that the data stream can't use `nested` field types. As we rely on auto-flattening mappings, there are also certain limitations that apply to integration and custom mappings for data streams that disable subobjects: https://www.elastic.co/guide/en/elasticsearch/reference/master/subobjects.html#_auto_flattening_object_mappings. Also, since auto-flattening will only available as of 8.14, the minimum stack version of that integration needs to be set to 8.14.

Reply:
I [expanded](https://github.com/elastic/package-spec/pull/727/commits/8353884278df16c27618fe0d5de42051726278cd) the subobjects description a little to:

- describe more precisely how subobjects work
- give an actual example
- list the caveats of enabling it at the data stream level

Reply:
Looks good 👍 
Maybe also clarify that it's still possible to send documents that have objects to a data stream that has set subobjects to false.

Reply:
Good point. 

I tried to clarify a couple of points.